### PR TITLE
#179: Added general telegram exceptions handling.

### LINKED ISSUE #179: Handle Telegram API response in case of failure
Bottery ignores errors from Telegram API. If, for example, an invalid token is used, no message is shown and, the process breaks because it tries to continue even if something is wrong with Telegram.

```
Traceback (most recent call last):
  File "bottery/telegram/engine.py", line 95, in polling
    tasks = [self.message_handler(msg) for msg in updates['result']]
KeyError: 'result'
```

### PR DESCRIPTION
**Pull request by ivkalita**
I added asyncio loop exception handler and Telegram exception handling on top of it.

Closes: #179

### CODE REVIEW

File: bottery/bottery.py [hunk 1]
```diff
@@ -80,8 +80,14 @@ def configure_server(self, port):
             port=port,
         ))
 
+    def exception_handler(self, loop, context):
+        click.echo(context.get('exception'))
+        loop.default_exception_handler(context)
+        loop.stop()
```
Comment (rougeth) [inline bottery/bottery.py]: Should we call `self.stop()` here?

File: bottery/telegram/engine.py [hunk 2]
```diff
@@ -85,6 +85,13 @@ def __init__(self, *args, **kwargs):
 
         updates = await self.api.get_updates(**payload)
 
+        if not updates.get('ok'):
+            msg = '[{}] Unable to receive updates, data={}, error'.format(
+                self.platform,
+                updates
+            )
+            raise Exception(msg)
```
Comment (rougeth) [inline bottery/telegram/engine.py]: Should we create a specific exception? Something like `PlatformError`? What do you think?

**Comment by rougeth** [comment]
Hi @kaduev13, thanks for the PR. I made some comments, could you take a look at them? Also, what do you think about adding some tests?

File: bottery/bottery.py (hunk 1 above)
Comment (kaduev13) [reply bottery/bottery.py]: Unfortunately, calling `self.stop()` here will cause an exception "cannot close running event loop". Also, it's semantically two different methods: `self.loop.close()` (inside `self.stop()`) tries to close event loop immediately (which is not possible in exception handler), but `loop.stop()` marks loop to be closed ASAP. I was thinking about replacing `self.loop.close()` 
 by `self.loop.stop()`, but I'm not sure if this issue a good place for such a proposal 😄

File: bottery/telegram/engine.py (hunk 2 above)
Comment (kaduev13) [reply bottery/telegram/engine.py]: Sure, I'll change an `Exception` to something more meaningful 😄

**Comment by kaduev13** [comment]
Hi @rougeth! Thank you for the review, I'll make the fixes tomorrow. About the tests – yeah, I'll add them, sorry for not doing it in the initial PR – I tried to find `polling` test case but found nothing and decided to send you a draft without tests to discuss it first.

File: bottery/bottery.py (hunk 1 above)
Comment (rougeth) [reply bottery/bottery.py]: I think `self.stop` wasn't  done the right way. There's another issue related to it (https://github.com/rougeth/bottery/issues/175). So yep, this PR isn't the best place to do discuss it indeed, but I think we should have at least a commend here saying that the stop/close should be done by another `self.stop()`.

Comment (kaduev13) [reply bottery/bottery.py]: Ok, so I'll add a comment in this PR, and also will try to fill a PR for the #175 to stop the event loop gracefully.

File: bottery/exceptions.py [hunk 3]
```diff
@@ -9,3 +9,13 @@ class ValidationError(Exception):
...
+
+
+class PlatformError(Exception):
+    def __init__(self, platform, message):
+        super().__init__()
+        self.platform = platform
+        self.message = message
+
+    def __str__(self):
+        return '[{}] {}'.format(self.platform, self.message)
```
Comment (rougeth) [inline bottery/exceptions.py:21]: I started questioning myself if the `[{}] {}` is a pattern that we should keep following. In this case, I think we don't need the `self.platform`, the exception itself should be able to give all the info we need. Maybe `class TelegramException(PlatformException)` is a good idea. What do you think?

**Pull request closed without merging**